src/Persistence.EventStore/EventStoreAggregateRepository.cs

```diff
@@ -1,31 +1,42 @@
-﻿using System;
-using System.Linq;
-using System.Text;
-using CR.AggregateRepository.Core;
-using CR.AggregateRepository.Core.Exceptions;
-using EventStore.ClientAPI;
-using EventStore.ClientAPI.Exceptions;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
+﻿// <copyright file="EventStoreAggregateRepository.cs" company="Cognisant">
+// Copyright (c) Cognisant. All rights reserved.
+// </copyright>
 
 namespace CR.AggregateRepository.Persistence.EventStore
 {
+    using System;
+    using System.Linq;
+    using System.Text;
+    using Core;
+    using Core.Exceptions;
+    using global::EventStore.ClientAPI;
+    using global::EventStore.ClientAPI.Exceptions;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    /// <inheritdoc />
+    /// <summary>
+    /// Implementation of <see cref="T:CR.AggregateRepository.Core.IAggregateRepository" /> which uses Event Store as underlying storage for an aggregate's events.
+    /// </summary>
     public class EventStoreAggregateRepository : IAggregateRepository
     {
-        private readonly IEventStoreConnection _connection;
         private const int ReadPageSize = 1000;
 
-        public EventStoreAggregateRepository(IEventStoreConnection connection)
-        {
-            _connection = connection;
-        }
+        private readonly IEventStoreConnection _connection;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventStoreAggregateRepository"/> class using the provided <see cref="IEventStoreConnection"/> to access and save events for an <see cref="IAggregate"/>.
+        /// </summary>
+        /// <param name="connection">The <see cref="IEventStoreConnection"/> to read from and write to.</param>
+        public EventStoreAggregateRepository(IEventStoreConnection connection) => _connection = connection;
 
+        /// <inheritdoc />
         public void Save(IAggregate aggregateToSave)
```

Comment:
We say `See implementation specific documentation for the circumstances in which this may be thrown.` on the base class for this but haven't actually specified on the implementation when it would be thrown.